### PR TITLE
Add Integration Tests for SAML ECP Profile

### PR DESCRIPTION
**Pull request by WinmaH**
When should this PR be merged

This PR should me merged after merging the following
https://github.com/wso2-extensions/identity-inbound-auth-saml/pull/196

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java [hunk 1]
```diff
@@ -365,6 +373,24 @@ public static HttpResponse sendGetRequest(String url, String userAgent, HttpClie
...
         return httpClient.execute(request);
     }
+    public static HttpResponse sendECPPostRequest(String url, String userAgent, HttpClient httpClient, String username, String password ,String soapRequest){
+        HttpPost request = new HttpPost(url);
+        HttpResponse response = null;
+        String auth = username + ":" + password;
+        byte[] encodedAuth = Base64.encodeBase64(
+                auth.getBytes(StandardCharsets.ISO_8859_1));
+        String authHeader = "Basic " + new String(encodedAuth);
+        System.out.println(authHeader);
```
Comment (WinmaH) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java]: remove the sout

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLECPSSOTestCase.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        }
+        this.config = config;
+    }
+
+    @DataProvider(name = "samlConfigProvider")
+    public static SAMLConfig[][] samlConfigProvider(){
+        return  new SAMLConfig[][]{
+                {new SAMLConfig(TestUserMode.SUPER_TENANT_ADMIN, User.SUPER_TENANT_USER, HttpBinding.HTTP_SOAP,
+                        ClaimType.LOCAL, App.ECP_APP)},
+                {new SAMLConfig(TestUserMode.SUPER_TENANT_ADMIN, User.SUPER_TENANT_USER, HttpBinding.HTTP_SOAP,
```
Comment (WinmaH) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLECPSSOTestCase.java]: remove duplicated configs

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java [hunk 3]
```diff
@@ -366,6 +374,24 @@ public static HttpResponse sendGetRequest(String url, String userAgent, HttpClie
         return httpClient.execute(request);
     }
 
+    public static HttpResponse sendECPPostRequest(String url, String userAgent, HttpClient httpClient, String username, String password ,String soapRequest){
+        HttpPost request = new HttpPost(url);
+        HttpResponse response = null;
+        String auth = username + ":" + password;
+        byte[] encodedAuth = Base64.encodeBase64(
+                auth.getBytes(StandardCharsets.ISO_8859_1));
```
Comment (omindu) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java]: Use UTF-8

**Comment by omindu** [comment]
Please reopen once issues in wso2-extensions/identity-inbound-auth-saml#196 are addressed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLECPSSOTestCase.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
+package org.wso2.identity.integration.test.saml;
```
[the remark this answers is not in the document]
Comment (WinmaH) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLECPSSOTestCase.java]: > Missing licence header

added

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java (hunk 3 above)
Comment (WinmaH) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java]: > Use UTF-8
Changed

**Pull request closed without merging**